boxlayout: handle widget removal

boz opened the pull request.

gdamore commented:
This looks good.  I guess I didn't really test that code before.  Can you describe your testing process for this?  I'm inclined to just merge this as is without further ado, but as no test case is presented, it would be helpful to understand that.

boz commented:
I'm tinkering with an [app](https://github.com/boz/kubetop) and just fixed it while debugging the problem.  I'm still in an exploratory phase and haven't been writing many tests.

The behaviour change that most needs to be baked in with a test is that all instances of `widget` are now removed.  The rest is a pretty obvious bugfix IMHO.

If I find some time I can write tests for this.  Not sure if/when that'll happen, unfortunately.
